engine/src/flutter/shell/platform/windows/display_monitor.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "display_monitor.h"
+
+#include <utility>
+#include <vector>
+
+#include "flutter/shell/platform/windows/dpi_utils.h"
+#include "flutter/shell/platform/windows/flutter_windows_engine.h"
+
+namespace flutter {
+
+DisplayMonitor::DisplayMonitor(
+    FlutterWindowsEngine* engine,
+    std::shared_ptr<WindowsProcTable> windows_proc_table)
+    : engine_(engine), win32_(std::move(windows_proc_table)) {
+  if (win32_ == nullptr) {
+    win32_ = std::make_shared<WindowsProcTable>();
+  }
+}
+
+DisplayMonitor::~DisplayMonitor() {}
+
+void DisplayMonitor::UpdateDisplays() {
+  std::vector<FlutterEngineDisplay> displays;
+
+  int display_count = win32_->GetSystemMetrics(SM_CMONITORS);
+
+  DISPLAY_DEVICE display_device = {0};
+  display_device.cb = sizeof(DISPLAY_DEVICE);
+
+  for (int i = 0; win32_->EnumDisplayDevices(nullptr, i, &display_device, 0);
```

Comment:
I think instead of `EnumDisplayDevices` we should do `EnumDisplayMonitors` instead. That gives you a list of `HMONITORS` directly, instead of having to filter the devices and using `MonitorFromPoint`.